### PR TITLE
Split choose service page into live services and trial mode services

### PR DESCRIPTION
**Pull request by quis**
# Why we’re changing this page 

At the moment all your services are jumbled together on the choose service page when logging in to Notify. In reality some services are much more important than others.

When we give people access to the organisation view, we don’t want to duplicate the list of live services on the choose service page. It would be confusing to have two pages which do sort of the same thing.

So this pull request makes two main changes:
- splits the list of services into two lists, live services and trial mode
- rolls up the live services list for users who belong to an organisation

More detail: https://www.pivotaltracker.com/story/show/166055997

# Before

image

# After 

image

# Use of API endpoints 

This pull request uses existing API endpoints, but abstracts them with the model layer. This means that, while the code works, it could get quite inefficient in extreme cases. For example for each organisation that someone belongs to we’re doing API call to `/organisation/<id>` and `/organisation/<id>/services`, because the existing `/user/<id>/organisation-and-services` endpoint doesn’t return all the information we need.

The subsequent work to update the endpoints to make them more efficient is in:
- https://github.com/alphagov/notifications-api/pull/2539
- https://github.com/alphagov/notifications-admin/pull/3012

### CODE REVIEW

**Comment by quis** [comment]
This does the same for the platform admin user page now too:

image

image

File: app/models/user.py [hunk 1]
```diff
@@ -237,6 +239,71 @@ def locked(self):
...
+        ]
+
+    @property
+    def live_services(self):
+        return [
+            service for service in self.services if service.live
+        ]
+
+    @property
+    def live_services_not_beloning_to_users_organisations(self):
```
Comment (klssmith) [inline app/models/user.py]: Typo here - should be 'belonging'

File: app/models/user.py [hunk 2]
```diff
@@ -251,6 +318,14 @@ def default_organisation_type(self):
             return 'nhs'
         return None
 
+    @property
+    def has_multiple_services_or_organisations(self):
```
Comment (klssmith) [inline app/models/user.py]: This name was slightly confusing

File: app/models/user.py [hunk 3]
```diff
@@ -251,6 +318,14 @@ def default_organisation_type(self):
...
+        return (
+            self.organisations or self.live_services
+        ) and (
+            self.trial_mode_services
+        )
```
[the remark this answers is not in the document]
Comment (quis) [reply app/models/user.py]: Two live services but no trial mode services still gets the simple, one column view of the page (as now). Renamed the property as per @klssmith’s suggestion in edba0c7f5

File: app/templates/views/choose-account.html [hunk 4]
```diff
@@ -1,43 +1,119 @@
...
+        show_heading=current_user.trial_mode_services,
+        organisations=current_user.organisations,
+        services=current_user.live_services_not_beloning_to_users_organisations
+      ) }}
+    {% else %}
+      {{ service_list(
+        heading='Live services',
+        show_heading=(current_user.trial_mode_services and current_user.live_services),
+        services=current_user.live_services
+      ) }}
```
[the remark this answers is not in the document]
Comment (quis) [reply app/templates/views/choose-account.html]: I was on board until `current_user.live_services_not_belonging_to_users_organisations if current_user.organisations else current_user.live_services` 🤨

File: app/models/user.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (quis) [reply app/models/user.py]: It’s mostly to cover local/preview/etc where we might have services without an org, or services which belong to an org that we’re not a member of (ie that we’ve manually assigned to test something).

**Pull request merged**